entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityDataServiceImpl.java

```diff
@@ -130,7 +145,9 @@ public void upsertEntities(Entities request, StreamObserver<Empty> responseObser
                   Collectors.toUnmodifiableMap(
                       entity -> this.entityNormalizer.getEntityDocKey(tenantId, entity),
                       Function.identity()));
+      java.util.Collection<Entity> existingEntities = getExistingEntities(entities.values());
```

Comment:
So if we assume doc store behaves consistently (discussed elsewhere), then we don't need this extra query, the result of the original docs returned on upsert should be sufficient.

Reply:
upsertEntities() doesn't return old documents.  Probably you meant `getAndUpsertEntities` method which was calling `bulkUpsertAndReturnOlderDocuments`. I have already reverted my change there. Please have a look.

Reply:
you're right, I was, carry on!